optuna/integration/mlflow.py

```diff
@@ -0,0 +1,141 @@
+import optuna
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Dict  # NOQA
+
+try:
+    import mlflow
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    mlflow = object
+
+
+def _check_mlflow_availability():
+    # type: () -> None
+
+    if not _available:
+        raise ImportError(
+            "MLflow is not available. Please install MLflow to use this "
+            "feature. It can be installed by executing `$ pip install "
+            "mlflow`. For further information, please refer to the installation guide "
+            "of MLflow. (The actual import error is as follows: " + str(_import_error) + ")"
+        )
+
+
```

Comment:
Please add the `experimental` decorator like [this](https://github.com/optuna/optuna/blob/2d5f24b06eed56ece72b8dfa878135bb4bb63779/optuna/integration/lightgbm_tuner/__init__.py#L23). I think we've better release this feature as an experimental version at first.

Reply:
Ok. Added it for version 1.3.0

Reply:
Since 1.3.0 is release today - should I change this to 1.4.0?

Reply:
Right; please update this to 1.4.0.

Reply:
Changed experimental version number to 1.4.0